src/lib/support/Span.h

```diff
@@ -178,6 +178,12 @@ class FixedSpan
     constexpr FixedSpan(std::array<U, N> & arr) : mDataBuf(arr.data())
     {}
 
+    template <class U, typename = std::enable_if_t<std::is_same<std::remove_const_t<T>, std::remove_const_t<U>>::value>>
+    constexpr FixedSpan(Span<U> & span) : mDataBuf(span.data())
+    {
+        VerifyOrDie(N == span.size());
+    }
```

Comment:
This means that when we define an interface as taking a FixedSpan, error checking can be pushed out of the implementation where there is knowledge of the required size, into any caller passing Span, which might not know the required size.  And if the caller passes a Span of the wrong size, the program will terminate.

At minimum, this constructor needs the explicit keyword so calling code explicitly and knowingly constructs FixedSpan from Span.

But I think I question the use of this at all.  If we want to pass in Span, why can't consuming interfaces have length-checking code in their implementations?